app/models/time_reg.rb

```diff
@@ -3,14 +3,26 @@ class TimeReg < ApplicationRecord
 
   belongs_to :membership
   belongs_to :assigned_task
+
   has_one :project, through: :assigned_task
   has_one :task, through: :assigned_task, source: :task
   has_one :user, through: :membership
+  has_one :client, through: :project
 
   validates :notes, length: { maximum: 255 }
   validates :minutes, numericality: { greater_than_or_equal_to: 0, less_than_or_equal_to: 1440 }
   validates :membership, presence: true
   validates :assigned_task, presence: true
   validates :assigned_task_id, presence: true
   validates :date_worked, presence: true
+
+  scope :with_report_scope, -> (client_ids, project_ids, user_ids, task_ids) {
```

Comment:
We usually don't suffix scopes with `_scope`. 

I'm struggling a bit to find a good name here though. That is always difficult. Maybe just `for_report`?

Feel free to disagree or come up with something else